jvm/src/main/kotlin/app/cash/trifle/Certificate.kt

```diff
@@ -22,6 +27,40 @@ data class Certificate internal constructor(
       version = CERTIFICATE_VERSION
     ).encode()
 
+  /**
+   * Verify that the provided certificate matches what we expected.
+   * It matches the CSR that we have and the root cert is what
+   * we expect.
+   *
+   * @param certificateRequest -  request used to generate this certificate
+   * @param certificateChain - list of certificates with each subsequent certificate signing the one
+   *   before it, starting with the parent of this certificate.
+   * @return - false if certificate is not signed by the given root certificate, or if the
+   *   information present doesn't match that of the certificateRequest.
+   *   //TODO(dcashman): Return errors specific to each type of failure so clients can remediate.
+   */
+  fun verify(
+    certificateRequest: CertificateRequest,
+    certificateChain: List<Certificate>,
+    rootCertificate: Certificate
+  ): Boolean {
+    // First check to see if the certificate chain matches
+    val validator = CertChainValidatorFactory.get(rootCertificate)
+    if (!validator.validate(listOf(this) + certificateChain)) return false
```

Comment:
I think rather than concatenate `this` with the rest of the certificate chain being passed in (also should we just name this as certificates?), just validate on the certificate chain itself since the validator would attempt to remove the anchor.

Reply:
discussed offline.  The API assumes that the calling object is the leaf certificate, and that the provided list does not include it.  Clarified with the method documentation.